File: 백준/Silver/19941. 햄버거 분배/햄버거 분배.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int k = Integer.parseInt(st.nextToken());
        int cnt = 0;
        char[] eat = br.readLine().toCharArray();
        for(int i=0;i<n;i++){
            if(eat[i]=='P'){
                for(int j = Math.max(0,i-k);j<Math.min(n,i+k+1);j++){
                    if(eat[j]=='H'){
                        cnt++;
                        eat[j]='0';
                        break;
                    }
                }
            }
        }
        System.out.println(cnt);

    }
}

File: 백준/Silver/19941. 햄버거 분배/햄버거 분배.py
import sys
input = sys.stdin.readline

n, k = map(int, input().split())
hambuger = list(input().strip())
cnt = 0

for i in range(n):
    if hambuger[i] == 'P':
        for j in range(max(0, i - k), min(n, i + k + 1)):
            if hambuger[j] == 'H': 
                cnt += 1
                hambuger[j] = '0'
                break

print(cnt)